sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Mode.scala

```diff
@@ -146,8 +114,86 @@ case class Mode(
   override def withNewInputAggBufferOffset(newInputAggBufferOffset: Int): Mode =
     copy(inputAggBufferOffset = newInputAggBufferOffset)
 
-  override def withNewChildrenInternal(newLeft: Expression, newRight: Expression): Expression =
-    copy(child = newLeft, deterministicExpr = newRight)
+  override def sql(isDistinct: Boolean): String = {
+    reverseOpt.map {
+      reverse =>
+        if (reverse) {
+          s"$prettyName() WITHIN GROUP (ORDER BY ${child.sql} DESC)"
```

Comment:
to double check, no other database support `mode(col1) within group (order by col2)`?

Reply:
I can't find the usage.